### PR TITLE
Enabling hivevar

### PR DESCRIPTION
**Pull request by PelleUllberg**
hivevar support in HiveShell

### CODE REVIEW

File: src/main/java/com/klarna/hiverunner/HiveServerContainer.java [hunk 1]
```diff
@@ -61,7 +63,7 @@ public CLIService getClient() {
      * @param testConfig Specific test case properties. Will be merged with the HiveConf of the context
      * @param context    The context configuring the HiveServer and it's environment
      */
-    public void init(Map<String, String> testConfig, HiveServerContext context) {
+    public void init(Map<String, String> testConfig, Map<String, String> vars, HiveServerContext context) {
```
Comment (Thoymas) [inline src/main/java/com/klarna/hiverunner/HiveServerContainer.java]: Missing javadoc.

File: src/main/java/com/klarna/hiverunner/HiveServerContainer.java [hunk 2]
```diff
@@ -184,6 +191,11 @@ public HiveConf getHiveConf() {
         return hiveServer2.getHiveConf();
     }
 
-
+    public VariableSubstitution getVariableSubstitution() {
```
Comment (Thoymas) [inline src/main/java/com/klarna/hiverunner/HiveServerContainer.java]: It's not here in the diff but there are other public methods in this class where you call new VariableSubstitution() without this guard. I'd guess you want to replace all calls to such instantiation with a call to this method instead?

File: src/main/java/com/klarna/hiverunner/builder/HiveShellBase.java [hunk 3]
```diff
@@ -63,11 +65,13 @@
                   List<HiveResource> resources,
                   List<String> scriptsUnderTest) {
         this.hiveServerContainer = hiveServerContainer;
-        this.props = props;
+        this.hiveConf = props;
```
Comment (Thoymas) [inline src/main/java/com/klarna/hiverunner/builder/HiveShellBase.java]: Minor: Rename the input parameter props to hiveConf.

File: src/test/java/com/klarna/hiverunner/HivePropertiesTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ */
+package com.klarna.hiverunner;
+
+import com.klarna.hiverunner.annotations.HiveSQL;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(StandaloneHiveRunner.class)
+public class HivePropertiesTest {
```
Comment (Thoymas) [inline src/test/java/com/klarna/hiverunner/HivePropertiesTest.java]: Rename to HiveVariablesTest. The word variables seems more commonly used for the concept than properties.

File: src/test/java/com/klarna/hiverunner/HivePropertiesTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    @HiveSQL(files = {}, autoStart = false)
+    public HiveShell shell;
+
+    @Test
+    public void substitutedVariablesShouldBeExpanded() {
+        shell.setProperty("origin", "spanish");
```
Comment (Thoymas) [inline src/test/java/com/klarna/hiverunner/HivePropertiesTest.java]: This applies to all calls in this test class. Why test using the deprecated method setProperty? For backwards compatibility I guess. Shouldn't there also be symmetric tests for setHiveConfValue()?

**Comment by patduin** [comment]
Hi,
What is the status of this PR? 
We've been trying out the branch ourselves and everything is working nicely. We'd love to have this in a release. 
Kind regards,
 Patrick (Hotels.com)

**Comment by PelleUllberg** [comment]
@patduin Just merged it

**Comment by patduin** [comment]
Thanks!

**Pull request merged**